src/MiddlewareListener.php

```diff
@@ -59,7 +59,11 @@ public function onDispatch(MvcEvent $event)
 
         $caughtException = null;
         try {
-            $return = $middleware(Psr7Request::fromZend($request), Psr7Response::fromZend($response));
+            $psr7Request = Psr7Request::fromZend($request);
+            foreach ($routeMatch->getParams() as $key => $value) {
```

Comment:
I think will be nice to inject also while `RouteMatch` (as we have it in expressive: https://github.com/zendframework/zend-expressive/blob/master/src/Application.php#L431-L435), not just individual matched params.